File: app/src/main/java/com/jamieadkins/gwent/card/list/CardListFragment.java
package com.jamieadkins.gwent.card.list;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.jamieadkins.gwent.Injection;
import com.jamieadkins.gwent.R;
import com.jamieadkins.gwent.data.interactor.CardsInteractorFirebase;

/**
 * UI fragment that shows a list of the users decks.
 */

public class CardListFragment extends BaseCardListFragment<CardsContract.View> {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getActivity().setTitle(getString(R.string.card_database));
    }

    @Override
    public void setupPresenter() {
        setPresenter(new CardsPresenter(Injection.INSTANCE.provideCardsInteractor(getContext())));
    }
}

File: app/src/main/java/com/jamieadkins/gwent/card/list/CardListFragment.kt
package com.jamieadkins.gwent.card.list

import android.os.Bundle

import com.jamieadkins.gwent.Injection
import com.jamieadkins.gwent.R

/**
 * UI fragment that shows a list of the users decks.
 */

class CardListFragment : BaseCardListFragment<CardsContract.View>() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activity?.title = getString(R.string.card_database)
    }

    override fun setupPresenter() {
        presenter = CardsPresenter(Injection.provideSchedulerProvider(),
                Injection.provideCardRepository(),
                Injection.provideUpdateRepository())
    }
}
